aptos-move/block-executor/src/view.rs

```diff
@@ -133,6 +143,118 @@ impl<'a, T: Transaction, X: Executable> ParallelState<'a, T, X> {
+            {
+                Ok((version, v)) => {
+                    let data_read = DataRead::Versioned(version, v.clone());
+
+                    assert_ok!(
+                        self.captured_reads.borrow_mut().capture_read(
+                            group_key.clone(),
+                            Some(resource_tag.clone()),
+                            data_read
+                        ),
+                        "Resource read in group recorded once: may not be inconsistent"
+                    );
+
+                    return Ok(GroupReadResult::Value(v.extract_raw_bytes()));
+                },
+                Err(Uninitialized) => {
+                    return Ok(GroupReadResult::Uninitialized);
+                },
+                Err(TagNotFound) => {
+                    let data_read = DataRead::Versioned(
+                        Err(StorageVersion),
+                        Arc::<T::Value>::new(TransactionWrite::from_state_value(None)),
+                    );
+                    assert_ok!(
+                        self.captured_reads.borrow_mut().capture_read(
+                            group_key.clone(),
+                            Some(resource_tag.clone()),
+                            data_read
+                        ),
+                        "Resource read in group recorded once: may not be inconsistent"
+                    );
+
+                    return Ok(GroupReadResult::Value(None));
+                },
+                Err(Dependency(dep_idx)) => {
+                    if !self.wait_for_dependency(txn_idx, dep_idx) {
+                        bail!("Interrupted as block execution was halted");
+                    }
+                },
+                Err(TagSerializationError) => {
+                    unreachable!("Reading a resource does not require tag serialization");
+                },
+            }
+        }
+    }
+
+    fn read_group_size(
+        &self,
+        group_key: &T::Key,
+        txn_idx: TxnIndex,
+    ) -> anyhow::Result<GroupReadResult> {
+        use MVGroupError::*;
+
+        if let Some(group_size) = self.captured_reads.borrow().group_size(group_key) {
+            return Ok(GroupReadResult::Size(group_size));
+        }
+
+        loop {
+            match self
+                .versioned_map
+                .group_data()
+                .get_group_size(group_key, txn_idx)
+            {
+                Ok(group_size) => {
+                    assert_ok!(
+                        self.captured_reads
+                            .borrow_mut()
+                            .capture_group_size(group_key.clone(), group_size),
+                        "Group size may not be inconsistent: must be recorded once"
+                    );
+
+                    return Ok(GroupReadResult::Size(group_size));
+                },
+                Err(Uninitialized) => {
+                    return Ok(GroupReadResult::Uninitialized);
```

Comment:
Same here. We should be fetching the data from storage if the data is not available in versioned_map right?

Reply:
see the caller.